coralproject/talk
# Optimize tag count resolution via caching

## What does this PR do?

Updates the logic for calculating the tag counts on a story to retrieve them from a cache of counts on the story document.

Also adds logic to generate the tag counts on the story document if they're not available (migrates existing stories).

## What changes to the GraphQL/Database Schema does this PR introduce?

No changes.

## Does this PR introduce any new environment variables or feature flags? 

No.

## If any indexes were added, were they added to `INDEXES.md`?

No new indexes.

## How do I test this PR?

- Go to a story that you have used before
- Load the page
- Open your MongoDB compass, VS Code Mongo tools, etc and find the story document
- Confirm that a new field for `tags` has shown up under `commentCounts` on the story document
- Check that the counts are correct for how many comments are featured, written by an admin, moderator, staff, etc
- Use a Q&A and a Ratings & Review story and confirm the tags are accurate there as well while reviewing, asking questions, answering, experts, etc... 
 
## How do we deploy this PR?

No special considerations to deploy.
